docs/guide.rst

```diff
@@ -31,108 +31,131 @@ The `WGPU <https://github.com/pygfx/wgpu-py>`_ library provides the low level AP
------------------------------
+    scene = gfx.Scene()
 
-Your visualization must end up on the screen somehow. For this, we use the
-canvas abstraction provided by wgpu-py.
+Right now this scene is very desolate. There is no light, no objects, and
+nothing that can look at those objects. Let's change this by adding some
+`light` and a `camera`::
 
-.. code-block:: python
-
-    # Create Qt widget that can function as a canvas
-    from wgpu.gui.qt import WgpuCanvas
-    canvas = WgpuCanvas()
+    # and god said ...
+    scene.add(gfx.AmbientLight())
 
+    camera = gfx.PerspectiveCamera(70, 16 / 9)
+    camera.position.z = 400
 
-We can ask the canvas to schedule a draw event, and tell it what to do
-to perform a draw.
+Now there is light and a camera to perceive the light. To complete the setup
+we also need to add an object to look at::
 
-.. code-block:: python
+    geometry = gfx.box_geometry(200, 200, 200)
+    material = gfx.MeshPhongMaterial(color=(1, 1, 0, 1))
+    cube = gfx.Mesh(geometry, material)
 
-    def animate():
-       ...  # we'll get to this
+    scene.add(cube)
 
+Objects are slightly more complicated than lights or cameras. They have a
+`geometry`, which controlls an object's form, and a `material`, which controls
+an object's appearance (color, reflectiveness, etc).
 
-    canvas.request_draw(animate)
+Now we have all the necessary ingredients and it is time to take a look. To do
+so we need to create a `canvas` to draw what we see (here a on-screen window)
+and a `renderer` that will look at the scene (through the camera) and draw what
+it sees onto the canvas::
 
+    from wgpu.gui.auto import WgpuCanvas, run
 
-Setting up a renderer
----------------------
+    renderer = gfx.renderers.WgpuRenderer(WgpuCanvas())
+    renderer.render(scene, camera)
 
-To render your scene to the canvas, you need a renderer. A renderer
-needs a target to render to (which can be a canvas or a texture).
-And finally, you need a camera to define the point of view to render the scene from.
+    run()
 
-.. code-block:: python
+.. image:: _static/guide_static_cube.png
 
-    # A renderer is associated with a canvas (or a texture) that it renders to
-    renderer = gfx.renderers.WgpuRenderer(canvas)
+Nice, we rendered our first scene! While this technically completes the hello
+world example, we can go one step further and add a bit of flare. We can add a
+`draw_function` which allows us to add custom logic into the rendering process::
 
-    # A camera defines the viewpoint and projection
-    camera = gfx.PerspectiveCamera(70, 16 / 9)
+    def animate():
+        # rotate the cube a bit
+        rot = gfx.linalg.Quaternion().set_from_euler(gfx.linalg.Euler(0.005, 0.01))
+        cube.rotation.multiply(rot)
+        # update the canvas
+        renderer.render(scene, camera)
+        renderer.request_draw()
 
-    ...
+    # make the cube pop
+    scene.add(gfx.DirectionalLight())
+    
+    # create a new window and new renderer
```

Comment:
I think this may be confusing. My suggestion would be to only show the new code here (also not the `run()`. People can see a few lines down how things are put together.

Reply:
Fair. I was thinking of users that open an interpreter and copy past the bits into it while reading the guide. Those users will have to define a new canvas after closing it each time; hence the duplication.

Happy to update though, if you think this is more confusing than helpful.

Reply:
Yeah, I figured you had an interactive session in mind, but that's not really clear here.

Actually, in an interactive session, you could add the light, define the animate function, call `request_draw` on it, and the original canvas would now show the cube rotating!

Reply:
@almarklein Oh, cool. In that case I think we should update to that and highlight it. Maybe we should even start with the renderer and a black screen and then make it come to life incrementally. What do you think?

Reply:
> Actually, in an interactive session, you could add the light, define the animate function, call request_draw on it, and the original canvas would now show the cube rotating!

I wanted to try this to see how to update the guide, but I didn't manage to make this work. `run()` is a blocking call and waits until the window is closed, so you can't modify the scene after calling it. 

At the same time, I don't get any render before calling `run()`. Instead, I get a light gray box and a friendly notification from Windows asking me if the program is stuck:

![image](https://user-images.githubusercontent.com/4402489/199826989-f6803734-88ca-47fd-91c4-e3a3e6ce5e58.png)

Reply:
To resolve this in an elegant way, I moved the `animate` extension into the `putting it all together` section. So now there is a step-by-step hello world example followed by a full example that adds a custom draw function. How does this look?

Reply:
If there is a window, but no event loop that keeps it alive, it looks unresponsive to the OS.

The `run()`  function starts the event loop and is blocking by default. If run from an IDE with interactivity set up (with the Qt event loop if using a Qt widget, or with asyncio when using glfw) then `run` does not block and things are truely interactive.

However, this event loop stuff is a bit of a mess in Python, so we cannot expect the user to have it set up ...

I like how you solved it.

Reply:
> If run from an IDE with interactivity set up (with the Qt event loop if using a Qt widget, or with asyncio when using glfw) then run does not block and things are truely interactive.

Hm, I was running it in a normal python shell you'd get by calling the python executable. I guess you were referring to an ipython shell?

In any case, if you like the current solution then we can simply keep it like this and sort out the event loop when it becomes relevant the next time.

Reply:
> I guess you were referring to an ipython shell?

Yes, for example, or any IDE that can do some event loop integration. But this is another topic indeed. Let's forget about interactive use for now and go with the current approach 👍